cluster.go

```diff
@@ -289,6 +312,74 @@ func parseSlots(slots RedisMessage, defaultAddr string) map[string]group {
 	return groups
 }
 
+// parseShards - map redis shards for each redis nodes/addresses
+// defaultAddr is needed in case the node does not know its own IP
+func parseShards(shards RedisMessage, defaultAddr string) map[string]group {
+	parseNodeEndpoint := func(msg map[string]RedisMessage) string {
+		var ip string
```

Comment:
I kept it the same as in slots although I think that endpoint should be used instead.
> endpoint: The preferred endpoint to reach the node, see below for more information about the possible values of this field.

Reply:
After revisiting the document, I noticed that this logic should be applied to the `endpoint` field, not the `ip` field.

Therefore, we need to change the parse function to be like this:

```go
func parseNodeEndpoint(msg map[string]RedisMessage, defaultAddr string, tls bool) string {
	endpoint := msg["endpoint"].string
	switch endpoint {
	case "":
		return defaultAddr
	case "?":
		return ""
	}

	port := msg["port"].integer
	if tls && msg["tls-port"].integer > 0 {
		port = msg["tls-port"].integer
	}

	return net.JoinHostPort(endpoint, strconv.FormatInt(port, 10))
}
```